addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -139,6 +135,7 @@ export class DiscussCoreCommon {
                 const thread = this.store.Thread.get({ model: "discuss.channel", id: payload.id });
                 if (thread) {
                     thread.is_pinned = false;
+                    thread.isLocallyPinned = false;
```

Comment:
This should be done in the `unpin` method instead (local state shouldn't be sync from server).